test/presenter/favorite_toggle_button_test.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:bloc_test/bloc_test.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:r6_moovie_app/presenter/bloc/favorites/favorite_bloc.dart';
+import 'package:r6_moovie_app/presenter/bloc/favorites/favorite_event.dart';
+import 'package:r6_moovie_app/presenter/bloc/favorites/favorite_state.dart';
+import 'package:r6_moovie_app/presenter/widgets/home/favorite_toggle_button.dart';
+
+class MockFavoriteEvent extends Fake implements FavoriteEvent {}
+
+class MockFavoriteBloc extends MockBloc<FavoriteEvent, FavoriteState>
+    implements FavoriteBloc {}
+
+void main() {
+  late FavoriteBloc mockFavoriteBloc;
+
+  setUp(() {
+    mockFavoriteBloc = MockFavoriteBloc();
+    when(() => mockFavoriteBloc.state).thenReturn(FavoritesInitialState());
+  });
+
+  Future<void> buildApp(
+      WidgetTester tester,
+      dynamic media,
+      bool isFavorite,
+      ) async {
+    await tester.pumpWidget(
+      MaterialApp(
+        home: BlocProvider<FavoriteBloc>.value(
+          value: mockFavoriteBloc,
+          child: FavoriteToggleButton(
+            media: media,
+            isFavorite: isFavorite,
+          ),
+        ),
+      ),
+    );
+    await tester.pump();
+  }
+
+  group('FavoriteToggleButton', () {
+    testWidgets('shows favorite icon when isFavorite is true',
+            (WidgetTester tester) async {
+          await buildApp(tester,anything, true);
+
+          expect(find.byIcon(Icons.favorite), findsOneWidget);
+          expect(find.byIcon(Icons.favorite_border), findsNothing);
+        });
+
+    testWidgets('shows border icon when isFavorite is false',
+            (WidgetTester tester) async {
+
+          await buildApp(tester, anything, false);
+
+          expect(find.byIcon(Icons.favorite_border), findsOneWidget);
+          expect(find.byIcon(Icons.favorite), findsNothing);
+        });
```

Comment:
Acredito que poderia criar uma função auxiliar que possa ser chamada dentro de ambos os teste, porém alterando apenas a mensagem. Ambos os testes estão testando a mesma "coisa". Poderia fazer por exemplo:

```suggestion
void testFavoriteIcon(WidgetTester tester, bool isFavorite, IconData expectedIcon, IconData unexpectedIcon) async {
//Implementação dos expects

}
        });
```

Essa função poderá ser chamada assim dentro "testWidgets": 
```suggestion
        testWidgets('shows favorite icon when isFavorite is true', (tester) async {
      testFavoriteIcon(tester, true, Icons.favorite, Icons.favorite_border);
    });
```